pkg/api/v1/clients/kube/resource_client_factory.go

```diff
@@ -97,6 +104,9 @@ func NewKubeSharedCacheForConfig(ctx context.Context, cluster string, restConfig
 // and, when started, creates a kubernetes controller that distributes notifications for changes to the watches that
 // have been added to the factory.
 // All direct operations on the ResourceClientSharedInformerFactory are synchronized.
+// Note it might be best to use this as a Singleton instead of a Structure
+// if a Singleton suggestion is proposed, then Register will no long be needed,
+// and we can just use Register New Namespace.  Also this would clear up some mutexes, that are being used.
```

Comment:
Do we want to consider this idea, or at least consider registering and starting the resource client factory on `NewFunction()`?